### ISSUE TITLE
Missing optimization

### ISSUE THREAD

**Issue by polluks** [cc65/cc65#2527]
Hi!
x ^= x;
should be equivalent to
x = 0;

**Comment by GorillaSapiens** [comment]
does anyone really do that on purpose though?  there are probably a ton of things like
x ^ 0 = x
x | 0 = x
x | x = x
x & 0xff = x
x & x = x
that could be checked and optimized.

**Comment by mrdudz** [comment]
> that could be checked and optimized, but you'd never actually see in real code.

BUT things like this could very well happen in intermediate versions of generated code. They should get optimized out either way :)

**Comment by Marrin** [comment]
> Well, did you ever see "+0"?

Yes. Can't recall the exact code but there was something like `registers+0`, `registers+1`, `registers+2` and also just `registers` to make it clear to the human reader when single register 0 was meant and when a pointer to registers that just happens to be also the pointer to the first register.

**Comment by mrdudz** [comment]
so after #2715, are there still simple cases that justifies keeping this ticket open?
